src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs

```diff
@@ -581,6 +581,38 @@ internal async Task<bool> CancelPreviousPoll(HttpContext context)
         }
     }
 
+    internal SetTransportState TrySetTransport(HttpTransportType transportType, HttpConnectionsMetrics metrics)
+    {
+        lock (_stateLock)
+        {
+            if (TransportType != HttpTransportType.None
+                && UseStatefulReconnect == false && TransportType != HttpTransportType.LongPolling)
```

Comment:
I notice that we use the same `UseStatefulReconnect == false && TransportType != HttpTransportType.LongPolling` condition in https://github.com/dotnet/aspnetcore/pull/50315/files#diff-1a6e4221ebf9343c0f831c3bddda91e203203f07d212575ad2f413ce3b3e2b34R301. Can we abstract this into an internal `!ClientReconnectExpected()` method on HttpConnectionContext? It'd be a lot more self-documenting.